app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantListViewModel.kt

```diff
@@ -35,26 +38,38 @@ class PlantListViewModel @ViewModelInject internal constructor(
     @Assisted private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
 
-    val plants: LiveData<List<Plant>> = getSavedGrowZoneNumber().switchMap {
-        if (it == NO_GROW_ZONE) {
-            plantRepository.getPlants()
-        } else {
-            plantRepository.getPlantsWithGrowZoneNumber(it)
-        }
+    private val growZoneChannel = ConflatedBroadcastChannel<Int>()
```

Comment:
Interesting pattern that may arise here. I wasn't sure what this `growZoneChannel` was doing at the beginning but if my assumptions are correct, this is just basically an observable type that we want to get from `SavedStateHandle` and because it doesn't support Flows yet, you need to retrieve it using `savedStateHandle.getLiveData(GROW_ZONE_SAVED_STATE_KEY, NO_GROW_ZONE)`. Is that correct?

For that, we could use the new APIs introduced in coroutines 1.4.0 (`StateFlow` and `SharedFlow`). As you need to access the current value of this for the `isFiltered()` function, the API you're looking for is `StateFlow`.

```
private val growZone = getLiveData(GROW_ZONE_SAVED_STATE_KEY, NO_GROW_ZONE)
                            .asFlow() // convert the LiveData to Flow
                            .stateIn(viewModelScope) // convert the Flow to StateFlow
```

That should also react for when you set a new grow zone.

Then, `plants` gets simplified to `growZone.flatMapLatest { ... }.asLiveData()`.

Reply:
Yes, we'll need to convert the `LiveData` from `SavedStateHandle` to Flow.  Thanks for the pointers!

I couldn't determine how to correctly use `stateIn` so I used an approach with `MutableStateFlow`.  I think this is a step towards that direction, with the next step collapsing `getSavedGrowZoneNumberOrDefault` into `growZone` - is that the correct?

Reply:
Conversation continued in subsequent thread.